feat: add 2 more liquidity fee setting methods (#2017)
vegaprotocol/specs: protocol/0042-LIQF-setting_fees_and_rewarding_lps.md

## Calculating liquidity fee factor

There are three ways to set the liquidity fee factor, and the default method is the 'Ascending Clock Auction'.

> ###  Ascending Clock Auction method 

Reviewer: I don't think this is the correct name for what we have. https://en.wikipedia.org/wiki/Japanese_auction

Reviewer: I think it's just the "marginal" method: https://en.wikipedia.org/wiki/Marginal_cost

Finally, we set the liquidity-fee-factor for this market to be the fee `LP-k-liquidity-fee-factor`.

> ### Example for fee setting mechanism in Ascending Clock Auction method

Reviewer: ```suggestion
### Example for fee setting mechanism using the marginal cost method
```

(average entry valuation) = 1090.9
```

> ###  "Vickrey-Clarke-Groves (VCG) auction" ("VCG mechanism") for setting the liquidity fee factor

Reviewer: I'd just call this "stake-weighted-average".

The liquidity fee factor is set as the weighted average of the liquidity fee factors, with weights assigned based on the supplied stake from each liquidity provider, which can also account for the impact of one supplier's actions on others.

> ###  Vickrey auction for setting the liquidity fee factor 

Reviewer: I don't think we want this one. Let's keep things simple.

(average entry valuation) = 1090.9
```

### Stake-weighted-average method for setting the liquidity fee factor

The liquidity fee factor is set as the weighted average of the liquidity fee factors, with weights assigned based on the supplied stake from each liquidity provider, which can also account for the impact of one supplier's actions on others.

### "Constant Liquidity Fee" Method

> The liquidity fee factor is set to a constant number

Reviewer: ```suggestion
The liquidity fee factor is set to a constant directly as part of the market proposal. 
```
